internal/character/march7th/skill.go

```diff
@@ -0,0 +1,93 @@
+package march7th
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Skill  = "march7th-skill"
+	E6heal = "march7th-e6hot"
+)
+
+type shieldState struct {
+	shieldPercentage float64
+	shieldFlat       float64
+}
+
+func init() {
+	modifier.Register(Skill, modifier.Config{
+		Duration: 3,
+		Stacking: modifier.Replace,
+		Listeners: modifier.Listeners{
+			OnAdd: func(mod *modifier.Instance) {
+				mod.Engine().AddShield(Skill, info.Shield{
+					Source: mod.Source(),
+					Target: mod.Owner(),
+					BaseShield: info.ShieldMap{
+						model.ShieldFormula_SHIELD_BY_SHIELDER_DEF: mod.State().(shieldState).shieldPercentage,
+					},
+					ShieldValue: mod.State().(shieldState).shieldFlat,
+				})
+				if mod.Engine().Stats(mod.Owner()).CurrentHPRatio() >= 30 {
+					mod.AddProperty(prop.AggroPercent, 5)
+				}
+			},
+			OnRemove: func(mod *modifier.Instance) {
+				mod.Engine().RemoveShield(Skill, mod.Owner())
+			},
+			OnPhase1: func(mod *modifier.Instance) {
+				march7th, _ := mod.Engine().CharacterInfo(mod.Source())
+				if march7th.Eidolon >= 6 {
```

Comment:
rather than calling `CharacterInfo`, I'd pass the heal stats as part of the state. If E6 is not enabled, just set the heal value and heal percent to 0 in the state